lib/scala/kg/src/main/scala/io/github/tetherlessworld/mcsapps/lib/kg/stores/postgres/PostgresKgQueryStore.scala

```diff
@@ -15,26 +15,67 @@ final class PostgresKgQueryStore @Inject()(configProvider: PostgresStoreConfigPr
   import profile.api._
 
   override def getNode(id: String): Option[KgNode] = {
-    runSyncTransaction((for { node <- nodes if node.id === id } yield node).result.headOption).map(_.toKgNode(List(), List()))
+    runSyncTransaction((for {
+      ((node, source), (_, nodeLabel)) <- nodes
+        .withSources
+        .join(nodes.withNodeLabels)
```

Comment:
I don't know Slick, but I suspect you want the results of the join combinator here rather the for expression.